lib/Dialect/XTen/Transforms/XTenNamePass.cpp

```diff
@@ -83,8 +83,8 @@ struct XTenNamePass : public XTenNameBase<XTenNamePass> {
 
       layerName = getLayerName(type.str(), layerToName[type.str()]);
       auto attr = StringAttr::get(module.getContext(), layerName);
-
-      op->setAttr(llvm::StringRef("layer_name"), attr);
+      if(generateLayerNameAttr && !op->hasAttr("layer_name"))
```

Comment:
The idea was to add an option to (true) overwrite existing `layer_name` settings (and this would be the default, to match the pass's current behavior) or (false) only add `layer_name` if there isn't one already.

The option added here seems to only either (false) effectively disable the pass or (true) `layer_name` add `layer_name` if there isn't one already.

Reply:
Added `overwriteExisting` flag and set it to True by default.